photon-lib/src/main/scala/com/linkedin/photon/ml/function/L2Regularization.scala

```diff
@@ -59,16 +67,22 @@ trait L2Regularization extends ObjectiveFunction {
       input: Data,
       coefficients: Coefficients,
       normalizationContext: BroadcastWrapper[NormalizationContext]): Double =
-    super.value(input, coefficients, normalizationContext) + l2RegValue(convertToVector(coefficients))
+    super.value(input, coefficients, normalizationContext) + l2RegValue(convertToVector(coefficients), interceptIndex)
```

Comment:
We shouldn't need to pass `interceptIndex` as it is a property of the trait, thus can already be accessed from `l2RegValue`.